conda/instructions.py

```diff
@@ -145,11 +148,29 @@ def execute_instructions(plan, index=None, verbose=False, _commands=None):
         if cmd is None:
             raise InvalidInstruction(instruction)
 
-        cmd(state, arg)
+        try:
+            cmd(state, arg)
+        except:
+            failed_instructions.append((cmd, state, arg))
 
         if (state['i'] is not None and instruction in progress_cmds
                 and state['maxval'] == state['i']):
             state['i'] = None
             getLogger('progress.stop').info(None)
 
+    double_failures = []
```

Comment:
Good idea to check for double failures! Just one idea about that:

This workaround is an iterative approach:
1. Run all pending commands.
2. If some commands fail, finish the others, then go to 1 with the failed commands.

The question here is when to stop this iteration. What about checking if the list of `failed_instructions` did not change? If there was no change during the last iteration step, it doesn't make sense to continue the iteration.